lib/view_component/collection.rb

```diff
@@ -1,22 +1,24 @@
-
 # frozen_string_literal: true
 
 module ViewComponent
   class Collection
+
     def render_in(view_context, &block)
-      as = @component.collection_parameter_name
+      iterator = ActionView::PartialIteration.new(@collection.size)
 
       @collection.map do |item|
-        @component.new(@options.merge(as => item)).render_in(view_context, &block)
+        content = @component.new(component_options(item, iterator)).render_in(view_context, &block)
+        iterator.iterate!
+        content
       end.join.html_safe
     end
 
     private
 
     def initialize(component, object, **options)
-      @component = component
+      @component  = component
       @collection = collection_variable(object || [])
-      @options = options
+      @options    = options
```

Comment:
Can we revert whitespace changes on lines you didn't touch. It just confused the git history and adds to confusion resolving merge conflicts etc.